src/service/test/protocol.rs

```diff
@@ -200,9 +200,16 @@ pub fn audio(path: &Path) -> Request<()> {
 		.unwrap()
 }
 
-pub fn thumbnail(path: &Path, pad: Option<bool>) -> Request<()> {
+pub fn thumbnail(path: &Path, size: Option<ThumbnailSize>, pad: Option<bool>) -> Request<()> {
 	let path = path.to_string_lossy();
 	let mut endpoint = format!("/api/thumbnail/{}", url_encode(path.as_ref()));
+	if let Some(s) = size {
+		match s {
+			ThumbnailSize::Small => endpoint.push_str("?size=small"),
```

Comment:
I think this will generate invalid URLs if combined with the pad option. It would end up looking like:
`?size=small?pad=true` instead of `?size=small&pad=true`. 

Less importantly, I have a personal preference for matching on the option with `Some(blah)` in the patterns rather than combining and `if let` with a `match`.

Reply:
Oh man, I don't know what I thought there.